src/sentry/api/serializers/models/group.py

```diff
@@ -713,7 +753,23 @@ def __init__(
         self.stats_period_end = stats_period_end
         self.matching_event_id = matching_event_id
 
+    def _get_seen_stats(self, item_list, user):
+        filtered_result = super(StreamGroupSerializerSnuba, self)._get_seen_stats(item_list, user)
+        # TODO: fix this hack. I would rather pass these vars down...but problems.
+        # At the very least, not losing their values after running this function would be nice.
+        self.snuba_filters = None
+        time_range_result = super(StreamGroupSerializerSnuba, self)._get_seen_stats(item_list, user)
+        self.start = None
+        self.end = None
+        lifetime_result = super(StreamGroupSerializerSnuba, self)._get_seen_stats(item_list, user)
```

Comment:
Should this function take the start,end,snuba_fiilters values as parameters that have a default of None. If the value is None they fall back to the value stored in the attribute? That would let you get around this problem.

An alternative approach could be to move the work done in the parent class' `_get_seen_stats()` method into a helper method that doesn't use any instance state and then call that method here.

Reply:
+1 on extracting to a helper function

Reply:
Brought it into a helper function. Going to try doing something similar for the get_stats -> query_tsdb chain.